storage/storage-interface/src/lib.rs

```diff
@@ -525,47 +536,62 @@ impl MoveStorage for &dyn DbReader {
         access_path: AccessPath,
         version: Version,
     ) -> Result<Vec<u8>> {
-        let (state_value, _) = self.get_state_value_with_proof_by_version(
-            &StateKey::AccountAddressKey(access_path.address),
-            version,
-        )?;
-        let account_state =
-            AccountState::try_from(&state_value.ok_or_else(|| {
-                format_err!("missing blob in account state/account does not exist")
-            })?)?;
+        let (state_value, _) = self
+            .get_state_value_with_proof_by_version(&StateKey::AccessPath(access_path), version)?;
 
-        Ok(account_state
-            .get(&access_path.path)
-            .ok_or_else(|| format_err!("no value found in account state"))?
-            .clone())
+        state_value
+            .ok_or_else(|| format_err!("no value found in DB"))?
+            .maybe_bytes
+            .ok_or_else(|| format_err!("no value found in DB"))
     }
 
     fn fetch_config_by_version(&self, config_id: ConfigID, version: Version) -> Result<Vec<u8>> {
-        let aptos_root_state = AccountState::try_from(
-            &self
-                .get_state_value_with_proof_by_version(
-                    &StateKey::AccountAddressKey(aptos_root_address()),
-                    version,
-                )?
-                .0
-                .ok_or_else(|| {
-                    format_err!("missing blob in account state/account does not exist")
-                })?,
-        )?;
-
-        match aptos_root_state.get(&access_path_for_config(config_id).path) {
-            Some(config) => Ok(config.to_vec()),
-            _ => aptos_root_state
-                .get(&dpn_access_path_for_config(config_id).path)
-                .map_or_else(
-                    || {
-                        Err(format_err!(
-                            "no config {} found in aptos root account state",
-                            config_id
-                        ))
-                    },
-                    |bytes| Ok(bytes.to_vec()),
-                ),
+        let config_value_option = self
+            .get_state_value_with_proof_by_version(
+                &StateKey::AccessPath(AccessPath::new(
+                    aptos_root_address(),
+                    access_path_for_config(config_id).path,
+                )),
+                version,
+            )?
+            .0;
+        match config_value_option {
```

Comment:
it's cumbersome to deal with the two `None` cases in  `Option<StateValue>`.. shall we create some sort of helper function for that??  For places where we don't care about the proofs, ideally the API can hide the details wrt "None as a placeholder for deleted value" situation..

Reply:
I like the idea of the API to simplify this. Can you clarify what you have in mind? Is this a helper function in StateValue, which takes `Option<StateValue>` and returns an `Option<Vec<u8>>`

Reply:
Yeah if you want to do it fancy: 

```
trait ToValueOption {
  pub fn to_value_opt()
}

impl ToValueOption for Option<StateValue> {
...
}
```

Reply:
Actually, nothing fancy is needed here.

```
  config_value_option
            .and_then(|x| x.maybe_bytes)
```
This will do.